src/main/java/java8/FunctionalInterfacesExample.java

```diff
@@ -0,0 +1,114 @@
+package java8;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.BiPredicate;
+import java.util.function.BinaryOperator;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.UnaryOperator;
+
+public class FunctionalInterfacesExample {
+	static Consumer<Student> consumer1 = (student) -> System.out.println(student.getId());
+	static Consumer<Student> consumer2 = (student) -> System.out.println(student.getName());
+	static List<Student> studentList = StudentDatabase.addStudents();
+	static Predicate<Integer> predicate1 = (i) -> i % 2 == 0;
+	static Predicate<Integer> predicate2 = (i) -> i % 5 == 0;
+	static Predicate<Student> pr1 = (student) -> student.getId() >= 2;
+	static Predicate<Student> pr2 = (student) -> student.getGrade() > 4;
+	static BiPredicate<Student, Student> bipr = (i, j) -> i.getGrade() < 4 && j.getId() == 4;
+	static BiPredicate<Student, Student> bipr1 = (i, j) -> i.getName() == "Ira" && j.getId() < 4;
+
+	public static void main(String args[]) {
+
+		// andThen method implementation
+		studentList.forEach(consumer1.andThen(consumer2));
+		// consumer implementation with condition
+		consumerWithCondition();
+		// BiConsumer Implementation
+		calculate();
+		// predicate functional interface
+		predicateAnd();
+		predicateOr();
+		predicateNegation();
+		predicateWithCondition();
+		// BiPredicate
+		biPredicate();
+		// Function interface
+		functionInterface();
+		// Supplier
+		supplier();
+
+	};
+
+	public static void consumerWithCondition() {
+		studentList.forEach(student -> {
+			if (student.getGrade() > 1) {
+				consumer1.andThen(consumer2).accept(student);
+			}
+			System.out.println(consumer1.toString());
```

Comment:
* Don't call to String method as it will be called by default if you are printing or using the object.